From 9d5627266d23ba41f98de72ffcdaa6a8a922d263 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Luttringer?= <seblu@seblu.net>
Date: Sat, 8 Apr 2023 12:57:52 +0200
Subject: [PATCH] Disable version check and error reporting

We do not want version to be checked by default, this is the job of pacman.
We do not want either have errors pushed to Scaleway sentry without proper
opt-in.
In order to avoid that we declare ourself has a non official release.
---
 internal/core/build_info.go | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/internal/core/build_info.go b/internal/core/build_info.go
index ad00e83e..31192891 100644
--- a/internal/core/build_info.go
+++ b/internal/core/build_info.go
@@ -43,10 +43,11 @@ const (
 )
 
 // IsRelease returns true when the version of the CLI is an official release:
-// - version must be non-empty (exclude tests)
-// - version must not contain metadata (e.g. '+dev')
+// Arch Linux release always returns false here in order to:
+// - Disable online version check. This is package manager job.
+// - Mandatory error reporting to Scaleway.
 func (b *BuildInfo) IsRelease() bool {
-	return b.Version != nil && b.Version.Metadata() == ""
+	return false
 }
 
 func (b *BuildInfo) GetUserAgent() string {
-- 
2.40.0

